src/core/global.c

```diff
@@ -1465,3 +1483,69 @@ void nn_global_rele_socket(struct nn_sock *sock)
     nn_sock_rele(sock);
     nn_glock_unlock();
 }
+
+int nn_setopt (int option, const void *optval, size_t optvallen)
+{
+    int idx;
+
+    switch (option) {
+    case NN_FORK_STRATEGY:
+        nn_assert(optvallen == sizeof (int));
+        idx = *(const int *) optval;
+        if (idx < 0 || idx >= NN_FORK_MAX_) {
+            errno = EINVAL;
+            break;
+        }
+#ifndef NN_HAVE_WINDOWS
+        nn_fork_strategy = &nn_fork_strategies[idx];
+#endif
+        return 0;
+    default:
+        errno = ENOPROTOOPT;
+        break;
+    }
```

Comment:
The break/return strategy is a little mixed here. Perhaps introduce `int rc` and just use breaks, with a single return at the end of the function?

That said ... not sure if `setopt` is a good idea yet, compared to say a build-time flag configuration option... thoughts?